src/mscorlib/src/System/Random.cs

```diff
@@ -114,11 +121,39 @@ public Random()
           return retVal;
       }
 
+
+      [ThreadStatic]
+      private static Random t_threadRandom;
+      private static Random s_globalRandom = new Random(GenerateGlobalSeed());
+
+      /*=====================================GenerateSeed=====================================
+      **Returns: An integer that can be used as seed values for consecutively
+                 creating lots of instances on the same thread within a short period of time.
+      ========================================================================================*/
+      private static int GenerateSeed() {
+          if (t_threadRandom == null) {
+              int seed;
+              lock (s_globalRandom) {
+                  seed = s_globalRandom.Next();
+              }
+              t_threadRandom = new Random(seed);
+          }
+          return t_threadRandom.Next();
+      }
+
+      /*==================================GenerateGlobalSeed====================================
+      **Action:  Creates a number to use as global seed.
+      **Returns: An integer that is safe to use as seed values for thread-local seed generators.
+      ==========================================================================================*/
+      private static int GenerateGlobalSeed() {
+          return Guid.NewGuid().GetHashCode();
```

Comment:
My only remaining issue with Guid is that `Guid.NewGuid().GetHashCode()` has no clear guarantees. Environment.TickCount does change reliably over time, but may introduces cross-process issues during a short interval (Do we want to care about those?). What do we know about the possibility of new guid hash codes to yield nearly identical values?
If we care about multiple processes, it maybe should be a Guid - hoping(?) that the chance of `GetHashCode()` returning **equal** numbers is low enough (for comparison: `Environment.TickCount` cycles after 49.8 days).